pkg/trantor/testing/smr_property_test.go

```diff
@@ -120,6 +120,12 @@ func checkEventTraces(eventLogFiles map[t.NodeID]string, transactionsSubmitted i
 
 	// Check if each node delivered as many transactions as were submitted.
 	for _, db := range deliveredBatches {
+
+		// Skip this check if no transactions were submitted.
+		if transactionsSubmitted == 0 {
+			break
+		}
```

Comment:
Curious about this, can you elaborate on how the changes on quorum size calculation causes an error here?

Reply:
I think it has to do with how rxgo works. Here I take an observable that contains all the delivered batches, map each batch to an int64 representing the number of included transactions, and then take the sum of all those numbers (I really like the concept of reactive programming):
```go
cnt, err := db.Map(func(_ context.Context, b interface{}) (interface{}, error) {
			return int64(len(b.(*batchfetcherpb.NewOrderedBatch).Txs)), nil
		}).SumInt64().Get()
```
This returns an objectIf there are only empty batches, this returns an object (`cnt`) and `cnt.V` is the value we're looking for (`cnt` holds additional information about potential errors). As long as there is even a single batch, this works (if it is empty, the value is 0). However, for some reason, if no batch has been delivered whatsoever, the above construct doesn't return a `cnt` object with the value 0, but `nil`. And then `cnt.V` segfaults.

This is a really small corner case and is not really relevant for tests that deliver any batches (even empty). But like this the error is handled properly at least, instead of the test segfaulting (since now we check whether `cnt == nil`). The check for `transactionsSubmitted == 0` is added since, technically, this particular test should pass even if no batches have been delivered at all.